PfamTests/data/HMM.3
HMMER3/a [3.0.dev | October 2008]
NAME  SEED
LENG  147
ALPH  amino
RF    no
CS    no
MAP   yes
DATE  Sun Nov 16 19:44:51 2008
COM   [1] /Users/rdf/Work/HMMER3/PreAlpha/hmmer-3.0.snap20081101/src/hmmbuild HMM.3 SEED.sto
NSEQ  5
EFFN  0.730286
STATS LOCAL     VLAMBDA 0.709747
STATS LOCAL         VMU -10.440608
STATS LOCAL        FTAU -4.221616
HMM          A        C        D        E        F        G        H        I        K        L        M        N        P        Q        R        S        T        V        W        Y   
            m->m     m->i     m->d     i->m     i->i     d->m     d->d
  COMPO   2.29721  4.29322  3.07171  2.71456  3.49325  3.20817  3.63640  2.94225  2.65796  2.40110  3.57967  3.17500  3.20412  3.00059  2.76311  2.64144  2.83951  2.71858  4.12690  3.58817
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.00000        *
      1   3.10036  4.62561  4.22278  3.83083  3.28636  3.95398  4.52940  2.47819  3.58943  1.85947  1.10396  4.08796  4.41777  3.96223  3.77562  3.46479  3.41000  2.50213  5.14676  3.92121      1 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
      2   2.31870  4.82685  2.45203  2.13611  4.10973  3.36664  3.64447  3.49296  2.46371  3.12182  3.94014  2.91040  3.80157  2.79588  2.93547  2.63046  2.50711  2.71587  5.36822  4.02103      2 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
      3   2.28721  5.31923  1.51655  1.87160  4.64138  3.25828  3.74078  4.10711  2.69172  3.64776  4.47739  2.74147  3.83696  2.89722  3.24934  2.77071  3.12626  3.70164  5.83252  4.37251      3 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
      4   2.66020  4.57126  3.31568  2.76032  3.78232  3.49307  3.74037  3.15994  2.43868  2.10064  3.72214  3.18523  3.91801  2.96622  2.09647  2.35298  2.90909  2.90546  5.12857  3.86054      4 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
      5   2.65345  4.27327  3.83549  3.32167  3.39392  3.63544  4.11727  2.43585  3.20311  1.59077  3.36053  3.61333  4.09247  3.51924  3.47435  2.41761  2.94138  1.97372  4.99723  3.76340      5 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
      6   1.95107  4.62326  3.00512  2.55409  4.12031  3.27919  3.74475  3.52872  2.21635  3.14781  3.97128  3.01610  2.88663  2.91738  2.95766  2.21370  2.82744  3.16057  5.38804  4.08206      6 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
      7   2.54337  3.28862  3.32339  2.77088  3.55758  3.46126  3.01162  2.93912  2.71426  2.34859  3.51620  3.19501  3.03752  3.04810  3.08865  2.41056  2.79502  2.69012  4.97247  3.71503      7 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
      8   3.33692  4.75372  4.44326  4.08645  3.20278  4.18189  4.69267  2.41821  3.85265  0.70212  3.18093  4.34650  4.59168  4.18288  4.00173  3.77517  3.62491  2.49618  5.13684  3.89586      8 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
      9   2.85017  5.12122  3.15887  2.56339  4.49406  3.54781  2.77302  3.89560  1.80565  3.36730  4.18541  3.03943  3.92335  2.05304  2.04728  2.83308  3.04407  3.54130  5.45871  4.19490      9 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     10   1.72922  4.40713  3.07011  2.75265  4.25598  3.09222  3.96638  3.67773  2.86354  3.32836  4.15225  2.39796  3.74687  3.17666  3.26016  1.74833  2.79446  3.21368  5.55904  4.27679     10 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     11   3.15582  4.53416  4.55521  4.17918  3.48527  4.23320  4.88025  0.90080  4.01035  2.04185  3.39900  4.40266  4.65166  4.34401  4.18915  3.74614  3.45320  1.86304  5.40697  4.15715     11 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     12   2.02478  4.30310  3.40245  2.96224  3.83715  3.23026  3.98351  3.10692  2.91735  2.32579  3.79307  3.27866  3.82716  3.25901  3.26030  1.66594  2.79538  2.79894  5.25904  4.00348     12 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     13   1.12121  4.16987  3.59944  3.29516  4.17341  3.02267  4.29257  3.28993  3.27834  3.18078  4.10444  3.41121  3.76273  3.59208  3.55640  2.48006  2.15616  2.88928  5.59455  4.39209     13 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     14   2.66429  4.88658  2.72358  2.39883  4.17853  3.31940  2.85581  3.72643  2.50316  3.29710  4.12225  2.07748  3.82451  2.86809  2.93766  2.02475  2.94191  3.35682  5.43782  4.05274     14 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     15   0.80748  4.27794  3.62963  3.43136  4.21610  3.09764  4.43382  3.36578  3.45185  3.23982  4.25645  3.53894  3.85213  3.77896  3.68961  2.62800  2.91649  2.99259  5.62735  4.44611     15 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     16   3.08852  5.19743  3.51415  2.83046  4.66511  3.69548  3.62487  4.00677  1.29532  3.43807  4.31072  3.22570  4.05983  2.75863  1.54176  3.08134  3.24968  3.68878  5.46929  4.31113     16 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     17   2.16078  4.83314  1.93026  2.36769  4.41614  2.53558  3.77158  3.86956  2.66346  3.44317  4.24959  2.87983  3.78203  2.93093  3.16009  2.22063  2.92084  3.43541  5.64385  4.26819     17 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     18   3.25044  4.64278  4.63823  4.09401  3.07923  4.40626  4.69957  2.23443  3.89360  1.15063  1.69260  4.37917  4.64439  4.10736  4.05580  3.76019  3.48182  2.34710  5.11536  3.99193     18 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     19   1.98240  4.26956  3.93514  3.40206  3.42328  3.69483  4.19682  2.32224  3.30294  2.21408  2.27565  3.68789  4.13977  3.59331  3.57503  3.02295  2.96341  1.88454  5.05780  3.85934     19 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     20   1.88891  4.67181  2.81519  2.10712  4.20950  3.24032  3.79282  3.60581  2.64936  3.24289  4.07606  2.97687  3.79803  2.97337  3.08754  1.95235  2.88103  3.22683  5.48735  4.16349     20 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     21   2.89012  4.29192  4.40078  3.84179  2.35929  4.01480  4.09847  1.78938  3.70649  1.68953  3.10822  3.98603  4.33188  3.86379  3.84036  3.32245  3.11798  2.31110  4.51900  2.49581     21 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     22   0.80748  4.27794  3.62963  3.43136  4.21610  3.09764  4.43382  3.36578  3.45185  3.23982  4.25645  3.53894  3.85213  3.77896  3.68961  2.62800  2.91649  2.99259  5.62735  4.44611     22 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     23   2.01151  4.56908  3.31488  2.76125  3.83309  3.47244  3.76301  3.13300  2.46587  2.41007  3.73860  3.18626  3.91161  2.97838  2.06417  2.76941  2.90279  2.87964  5.18165  3.92182     23 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     24   1.51287  4.17823  3.45906  3.17900  4.29424  2.96118  4.24525  3.65494  3.22762  3.39409  4.24038  3.32649  3.71214  3.52261  3.53281  1.36722  2.72912  3.13453  5.65320  4.43302     24 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     25   3.19151  5.01450  3.69904  3.18029  4.40958  3.64574  3.95006  3.96447  2.31634  3.45171  4.45000  3.52675  4.14264  3.16448  0.76138  3.26942  3.44797  3.68213  5.42962  4.30493     25 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     26   2.87346  4.63053  3.57946  3.50032  4.63470  0.52170  4.60851  4.27521  3.70299  3.90871  4.88757  3.74413  4.02662  4.01481  3.91931  3.05032  3.36704  3.77295  5.68554  4.74281     26 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     27   1.06583  4.18818  3.45867  3.22536  4.33210  2.16794  4.30150  3.67681  3.31358  3.43038  4.28605  3.36037  3.72885  3.59503  3.60445  2.44924  2.76106  3.15592  5.68048  4.48783     27 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     28   2.61601  4.75956  3.00169  2.47030  3.99587  3.42678  3.63123  3.40843  2.13968  3.01868  3.84713  2.97581  3.82641  2.47869  2.80343  2.38015  2.50782  3.09846  5.26115  3.11955     28 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     29   2.55254  4.69221  2.89444  2.48912  4.13458  2.68036  2.83916  3.57859  2.54864  3.18233  4.00326  2.97310  2.49157  2.89838  2.98506  2.23960  2.85077  3.20955  5.40366  4.07343     29 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     30   2.11486  4.22257  3.60343  3.09739  3.59255  3.38962  3.99959  2.00615  3.03692  2.61266  3.54250  3.40147  3.91349  3.34303  3.35475  2.31416  2.37556  2.46687  5.06508  3.84281     30 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.23109  4.03092  1.66827  0.61958  0.77255  0.48576  0.95510
     31   2.70063  4.46388  3.30361  3.13742  4.13484  3.15369  4.21765  3.67058  3.19142  3.30153  4.35014  3.44782  0.90535  3.58144  3.44093  2.86702  3.13920  3.32045  5.33175  4.25518     31 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.03270  3.83253  4.55487  0.61958  0.77255  0.38957  1.13118
     32   2.31412  4.81378  2.95453  2.43996  4.15397  3.37425  3.63400  3.57561  2.37194  3.15548  3.96386  2.94657  3.01734  2.44785  2.45054  2.30028  2.84668  3.22443  5.36877  4.03887     32 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     33   2.15099  4.64919  2.93116  2.56659  4.15321  3.27449  3.78692  3.55705  2.58062  3.17837  4.02898  3.02587  2.06537  2.41872  2.97462  2.64147  2.88410  3.19598  5.43063  4.12515     33 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     34   2.58189  4.28208  3.48515  2.92122  3.39630  3.56473  3.82147  2.41400  2.84816  2.44989  3.37373  3.31659  3.94566  2.67572  3.18625  2.81848  2.47743  2.25393  4.86311  2.95262     34 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     35   2.65452  4.88710  2.92463  2.44636  4.24295  2.80043  3.63900  3.66731  2.32320  3.22734  4.03915  2.95148  2.53395  2.41870  2.33072  2.67147  2.90096  3.31057  5.41649  4.09260     35 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     36   2.13333  4.34287  3.36212  2.85894  3.74229  3.32841  3.88314  3.01046  2.81130  2.36276  3.66897  3.23391  2.86275  3.14790  3.17783  2.65108  2.08094  2.73320  5.15587  3.90824     36 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     37   1.88090  4.18734  3.49289  3.13125  4.16868  3.01271  4.16470  3.49130  3.13014  3.24016  4.08921  3.31542  3.72690  3.43632  3.45035  1.96946  1.51965  3.03321  5.53531  4.31948     37 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     38   2.26785  4.65518  2.59741  2.49642  3.94894  3.37257  3.68388  3.32223  2.52502  2.58010  3.81692  2.99174  3.07464  2.86182  2.97225  2.63124  2.44027  3.01335  5.26012  3.94722     38 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     39   2.67315  4.26769  3.85340  3.31544  3.39293  3.69201  4.11663  2.37291  3.21085  2.23567  2.65806  3.62548  3.02689  3.51296  3.48918  3.00494  2.94219  1.57556  4.99551  3.78668     39 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     40   2.65019  4.83939  2.94935  2.46201  4.13775  3.40708  2.53695  3.56188  2.07577  3.14550  3.96756  2.96551  3.83348  2.79331  2.73855  2.33214  2.50668  3.22771  5.35264  4.03195     40 - -
          2.68621  4.42096  2.77523  2.73126  3.46357  2.40516  3.72497  3.29357  2.67744  2.69358  4.24693  2.90350  2.73718  3.18149  2.89804  2.37890  2.77523  2.98521  4.58480  3.61506
          0.13644  2.12932  4.75326  0.65742  0.73019  0.48576  0.95510
     41   3.33995  4.69225  4.62352  4.18195  1.27299  4.34810  4.08814  2.51285  4.02747  1.52195  3.19070  4.25552  4.64091  4.13849  4.13848  3.72844  3.57993  2.57838  4.31376  2.72707     43 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     42   2.28796  5.32059  1.52915  1.85694  4.64178  3.25878  3.73930  4.10842  2.68892  3.64771  4.47651  2.74115  3.83649  2.89525  3.24640  2.76971  3.12498  3.70246  5.83174  4.37156     44 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     43   2.61234  4.64193  2.95834  2.22335  3.96290  3.37033  3.76569  3.25590  2.60288  2.97188  3.85784  3.04390  2.16259  2.96481  3.01227  2.70087  2.89099  2.45290  5.30375  4.00230     45 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     44   3.33692  4.75372  4.44326  4.08645  3.20278  4.18189  4.69267  2.41821  3.85265  0.70212  3.18093  4.34650  4.59168  4.18288  4.00173  3.77517  3.62491  2.49618  5.13684  3.89586     46 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     45   2.69075  4.50887  3.44885  3.07681  3.68914  3.43376  4.06939  3.02217  2.96586  2.10487  3.76755  3.42553  1.46944  3.38003  3.26655  2.87578  3.03812  2.82698  5.20877  3.91122     47 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     46   2.46958  4.38072  3.28331  2.78636  3.79538  3.30495  3.84259  3.14118  2.74626  2.42064  3.72119  3.17928  2.92437  3.08928  3.12474  2.24179  2.05821  2.84037  5.17918  3.91972     48 - -
          2.68618  4.42225  2.77519  2.73123  3.46354  2.40513  3.72494  3.29354  2.67741  2.69355  4.24690  2.90347  2.73739  3.18146  2.89801  2.37887  2.77519  2.98518  4.58477  3.61503
          0.02674  4.03092  4.75326  0.61958  0.77255  0.48576  0.95510
     47   2.97466  4.70285  3.59940  3.46321  4.44052  3.38152  4.52554  4.02933  3.53459  3.63792  4.69623  3.75275  0.57946  3.91851  3.76001  3.14768  3.42920  3.65619  5.58109  4.56417     49 - -

trunk/Changes
This file documents the revision history for Perl extension PfamViewer.

0.01  2009-01-26 15:48:38
        - initial revision, generated by Catalyst


PfamLib/Bio/Pfam/HMM/HMM.pm
# HMM.pm
#
# Author:        finnr
# Maintainer:    $Id: HMM.pm,v 1.1 2009-10-08 12:27:28 jt6 Exp $
# Version:       $Revision: 1.1 $
# Created:       Nov 24, 2008
# Last Modified: $Date: 2009-10-08 12:27:28 $
=head1 NAME

Template - a short description of the class

=cut

package Bio::Pfam::HMM::HMM;

=head1 DESCRIPTION

A more detailed description of what this class does and how it does it.

$Id: HMM.pm,v 1.1 2009-10-08 12:27:28 jt6 Exp $

=head1 COPYRIGHT

File: HMM.pm

Copyright (c) 2007: Genome Research Ltd.

Authors: Rob Finn (rdf@sanger.ac.uk), John Tate (jt6@sanger.ac.uk)

 This is free software; you can redistribute it and/or
 modify it under the terms of the GNU General Public License
 as published by the Free Software Foundation; either version 2
 of the License, or (at your option) any later version.
 
 This program is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.
 
 You should have received a copy of the GNU General Public License
 along with this program; if not, write to the Free Software
 Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 or see the on-line version at http://www.gnu.org/copyleft/gpl.txt
 
=cut

use strict;
use warnings;

use Moose;
use Moose::Util::TypeConstraints;
use Carp;
use Data::Dumper;

#-------------------------------------------------------------------------------

=head1 METHODS

=cut



subtype 'hmmVersion'
  => as Str
  => where { $_ =~ m/^HMMER3\/[ab]\s+\[.*\]/ }
  => message { "|$_| does not look like as HMMER3 version" };

subtype 'hmmName'
  => as Str
  => where { $_ =~ m/\S{1,15}/ }
  => message { "|$_| does not look like Pfam name or SEED" };

subtype 'hmmAcc'
  => as Str
  => where { $_ =~ m/PF\d{5}/ }
  => message { "|$_| does not look like Pfam accession" };


subtype 'hmmAlpha'
  => as Str
  => where { $_ eq 'amino' or $_ eq 'nucleic'  }
  => message { "|$_| does not look like a HMMER3 alphabet" };


subtype 'hmmMsvStats'
  => as HashRef
  => where { defined ($_->{mu}) and defined ($_->{lambda}) and ($_->{lambda} <= 0.8) and ($_->{lambda} >= 0.5) }
  => message { "Mu |$_->{mu}| and lambda |$_->{lambda}| must be defined and lambda must be between 0.5 and 0.8" };

  
subtype 'hmmViterbiStats'
  => as HashRef
  => where { defined ($_->{mu}) and defined ($_->{lambda}) and ($_->{lambda} <= 0.8) and ($_->{lambda} >= 0.5) }
  => message { "Mu |$_->{mu}| and lambda |$_->{lambda}| must be defined and lambda must be between 0.5 and 0.8" };

subtype 'hmmForwardStats'
  => as HashRef
  => where { defined ($_->{tau}) and defined ($_->{lambda}) and ($_->{lambda} <= 0.8) and ($_->{lambda} >= 0.5) }
  => message { "Tau |$_->{tau}| and lambda |$_->{lambda}| must be defined and lambda must be between 0.5 and 0.8" };
  
has 'version' => (
  isa      => 'hmmVersion',
  is       => 'rw',
  required => 1
);

has 'name' => (
  isa => 'hmmName',
  is  => 'rw',
  required => 1
);

has 'accession' => (
  isa => 'hmmAcc',
  is  => 'rw'
);

has 'description' => (
  isa => 'Str',
  is  => 'rw'
);

has 'length' => (
  isa  => 'Int',
  is   => 'rw',
  required => 1
);

has 'alpha' => (
  isa  => 'hmmAlpha',
  is   => 'rw',
  required => 1,
);

has 'rf' => (
  isa  => 'Bool',
  is   => 'rw',
  required => 1
);

has 'cs' => (
  isa  => 'Bool',
  is   => 'rw',
  required => 1
);

has 'map' => (
  isa  => 'Bool',
  is   => 'rw',
  required => 1
);

has 'date' => (
  isa  => 'Str',
  is   => 'rw',
  required => 1
);

has 'buildLine' => (
  isa => 'HashRef[Str]',
  is  => 'rw',
  required => 1,
  default => sub { {} },
);

has 'searchMethod' => (
  isa => 'Str',
  is  => 'rw',
);

has 'nSeq' => (
  isa  => 'Int',
  is   => 'rw',
  required => 1
);

has 'msvStats' => (
  isa => 'hmmMsvStats',
  is => 'rw',
  required => 1
);

has 'viterbiStats' => (
  isa => 'hmmViterbiStats',
  is => 'rw',
  required => 1
);

has 'forwardStats' => (
  isa => 'hmmForwardStats',
  is => 'rw',
  required => 1
);


has 'effn' => (
  isa  => 'Num',
  is   => 'rw',
  required => 1
);

has 'cksum' => (
  isa => 'Int',
  is  => 'rw',
  required => 1
);

has 'seqGA' => (
  isa => 'Num',
  is  => 'rw',
);

has 'domGA' => (
  isa => 'Num',
  is  => 'rw',
);

has 'seqTC' => (
  isa => 'Num',
  is  => 'rw',
);

has 'domTC' => (
  isa => 'Num',
  is  => 'rw',
);

has 'seqNC' => (
  isa => 'Num',
  is  => 'rw',
);

has 'domNC' => (
  isa => 'Num',
  is  => 'rw',
);

has 'emissionLines' => (
  isa => 'ArrayRef[ArrayRef]',
  is  => 'rw',
  default => sub { [] },
);

has 'mapPos'=> (
  isa => 'ArrayRef[Int]',
  is  => 'rw',
  default => sub{ [] }
);

has 'compLines' => (
  isa => 'ArrayRef[Str]',
  is  => 'rw',
  default => sub { [] },
);

__PACKAGE__->meta->make_immutable;
1;

